src/feature_filter/convert.ts

```diff
@@ -128,7 +128,7 @@ function runtimeTypeChecks(expectedTypes: ExpectedTypes): ExpressionFilterSpecif
 function convertComparisonOp(property: string, value: any, op: string, expectedTypes?: ExpectedTypes | null): ExpressionFilterSpecification {
     let get;
     if (property === '$type') {
-        return [op, ['geometry-type'], value] as ExpressionFilterSpecification;
+        return convertInOp('$type', [value], op === '!=');
```

Comment:
Can you elaborate a bit on this change and why it is needed?

Reply:
That's the essence of the PR.
The `Multi` and non-`Multi` variants of `geometry-type` must be considered when comparing with the fewer possible values of `$type`. Later on, `==` and `!=` expressions are transformed into `in` and `!in` expssions.
For example, `["``",  "$type", "LineString` ]` becomes `["in", ["geomerty-type"], ["LineString", "MultiLineString"]]`

Reply:
Doesn't this mean that it will convert `geomerty-type` to "old style" expression and there won't be a way to use it with new expressions?

Reply:
Apparently it upgrades the expression to new style. So no problem here.